server/middleware/token.js

```diff
@@ -101,24 +111,26 @@ function token(options) {
     assert(typeof TokenModel === 'function',
       'loopback.token() middleware requires a AccessToken model');
 
+    var err = null;
     if (req.accessToken !== undefined) {
       if (!enableDoublecheck) {
         // req.accessToken is defined already (might also be "null" or "false") and enableDoublecheck
         // has not been set --> skip searching for credentials
-        rewriteUserLiteral(req, currentUserLiteral);
-        return next();
+        err = rewriteUserLiteral(req, currentUserLiteral);
+        return next(err);
```

Comment:
I considered that initially but didn't like transferring responsibility to call `next()` to rewriteUserLiteral because it wasn't always the last call. Since we've refactored the findForRequest I think it makes more sense to pass `next in now.